loader/loader.c

```diff
@@ -5171,6 +5243,21 @@ VKAPI_ATTR VkResult VKAPI_CALL terminator_CreateInstance(const VkInstanceCreateI
     icd_create_info.enabledLayerCount = 0;
     icd_create_info.ppEnabledLayerNames = NULL;
 
+    uint32_t required_extra_instance_ext_count = 0;
+    res = loader_get_layer_required_instance_extensions(&required_extra_instance_ext_count, NULL);
+    if (res != VK_SUCCESS) {
+        loader_log(ptr_instance, VULKAN_LOADER_ERROR_BIT, 0,
+                   "terminator_CreateInstance: Failed to get required extra instance extension count from layers");
+        goto out;
+    }
+    char **required_extra_instance_exts = loader_stack_alloc(required_extra_instance_ext_count * sizeof(char *));
+    res = loader_get_layer_required_instance_extensions(&required_extra_instance_ext_count, required_extra_instance_exts);
```

Comment:
While I appreciate your usage of the Vulkan idiom to call once to get the count and then again to get the data, that isn't necessary *inside* the loader. We can pass allocated arrays back up the callstack.

Stack allocation is generally avoided where possible, and while we do use it in a handful of places in the loader, this isn't necessarily one of them.

Alternatively, there could be a `loader_get_layer_required_instance_extensions_length()` function to get the count of extensions, the `vkCreateInstance` allocates (with stack alloc) enough space for the layer required extensions + application enabled, then a second function fills in the extensions.